docker/Dockerfile.base

```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:xenial
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Do not add more stuff to this list that isn't small or critically useful.
+# If you occasionally need something on the container do
+# sudo apt-get update && apt-get whichever
+
+# hadolint ignore=DL3005,DL3008
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y \
+      ca-certificates \
+      curl \
+      iptables \
+      iproute2 \
+      iputils-ping \
+      knot-dnsutils \
+      netcat \
+      tcpdump \
+      net-tools \
+      lsof \
+      linux-tools-generic \
+      sudo \
+   && apt-get upgrade -y \
```

Comment:
Minor nits. Line 23 could prevent repeatability of builds and it will get the latest versions. Line 24 not needed for Ubuntu images.